p2p/http/libp2phttp.go

```diff
@@ -657,12 +660,22 @@ func (h *Host) RoundTrip(r *http.Request) (*http.Response, error) {
 		return nil, err
 	}
 	addr, isHTTP := normalizeHTTPMultiaddr(addr)
+	parsed, err := parseMultiaddr(addr)
```

Comment:
We need this parsing only if it's a http multiaddr, right? Otherwise we can just use the `addr`? 

I would prefer calling this only when it's an `/http` multiaddr because `parseMultiaddr` looks specific to parsing `/http` multiaddrs

Reply:
> We need this parsing only if it's a http multiaddr, right? 

Both. We use the http-path, host, and port in both code paths.

Would you prefer we name this something else?

Reply:
You only need `parsed.peer` in that code path so I was wondering if we should just get the peerID from the addr in that code path.  

This is fine too though.

Reply:
We need http path as well